src/main/java/message/Message.java

```diff
@@ -0,0 +1,46 @@
+package message;
+
+import java.util.Arrays;
+
+/**
+ * Encapsulates the base output message that Duke replies to the user.
+ * It includes methods to format the output message.
+ */
+public class Message {
+    private final String[] texts;
+
+    /**
+     * Instantiates a Message holding texts to be output to the user.
+     *
+     * @param texts Texts are vararg of strings.
+     */
+    public Message(String ...texts) {
+        assert texts.length > 0 : "A message should have least one string";
+
+        this.texts = texts;
+    }
```

Comment:
I think it would provide clarity to add in throw AssertionError to indicate there could be potential errors to catch.

Reply:
Would it be necessary to throw exceptions for assertions? Because assertions are used to catch programmer mistakes and are separate from exceptions. I would think that if an AssertionError happens, it should be picked up during development, be fixed and not allowed to pass to production? Additionally, they can be disabled?